### PR TITLE
Fix warnings related to nullability pointers

### PR DESCRIPTION
**Pull request by erychagov**
Fixed the following warning: "implicit conversion from nullable pointer to non-nullable pointer". 

Looks like in function fileWithId: (MSStorageBucket.m) we always can find a file. Or sometimes we can't? If not I'll change the code

### CODE REVIEW

File: MobileCenter/MobileCenter/Internals/Channel/MSChannelDefault.m [hunk 1]
```diff
@@ -264,9 +264,8 @@ - (void)startTimer {
 }
 
 - (void)resetTimer {
-  if (self.timerSource) {
+  if(self.timerSource) {
```
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/Internals/Channel/MSChannelDefault.m]: Have you checked for any side-effects on making the timersource `_Nonnull`?

File: MobileCenter/MobileCenter/Internals/Storage/MSStorageBucket.m [hunk 2]
```diff
@@ -20,7 +20,7 @@ - (MSFile *)fileWithId:(NSString *)fileId {
     results = [self.availableFiles filteredArrayUsingPredicate:predicte];
   }
 
-  return results.lastObject;
+  return (MSFile *_Nonnull) results.lastObject;
```
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/Internals/Storage/MSStorageBucket.m]: Are there any cases where `results.lastObject` can be `nil`?

File: MobileCenter/MobileCenter/Internals/Vendor/Reachability/MS_Reachability.m [hunk 3]
```diff
@@ -66,7 +66,13 @@ @implementation MS_Reachability {
 
 + (instancetype)reachabilityWithHostName:(NSString *)hostName {
   MS_Reachability *returnValue = NULL;
-  SCNetworkReachabilityRef reachability = SCNetworkReachabilityCreateWithName(NULL, [hostName UTF8String]);
+
+  const char * utf8HostName = [hostName UTF8String];
```
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/Internals/Vendor/Reachability/MS_Reachability.m]: Please revert the changes in MS_Reachability.m. It's based on Apple's sample code and I'd rather not change it. You can exclude this file from this kind of warnings.
I didn't mention this in my ticket for the task, so apologies, it's my fault.

File: MobileCenterCrashes/MobileCenterCrashes/Internals/MSCrashesCXXExceptionWrapperException.m [hunk 4]
```diff
@@ -12,7 +12,8 @@ - (instancetype)initWithCXXExceptionInfo:(const MSCrashesUncaughtCXXExceptionInf
   extern char *__cxa_demangle(const char *mangled_name, char *output_buffer, size_t *length, int *status);
   char *demangled_name = &__cxa_demangle ? __cxa_demangle(info->exception_type_name ?: "", NULL, NULL, NULL) : NULL;
 
-  if ((self = [super initWithName:[NSString stringWithUTF8String:demangled_name ?: info->exception_type_name ?: ""]
+  //stringWithUTF8String never return null for us because we always send correct string
```
Comment (TroubleMakerBen) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/MSCrashesCXXExceptionWrapperException.m]: `return` -> `returns`

`correct string` -> `a correct string`

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorAttachment.h [hunk 5]
```diff
@@ -1,5 +1,7 @@
 #import <Foundation/Foundation.h>
 
+NS_ASSUME_NONNULL_BEGIN
```
Comment (TroubleMakerBen) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorAttachment.h]: This is a model object, we don't need the `NS_ASSUME_NONNULL_BEGIN` here.

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorBinaryAttachment.h [hunk 6]
```diff
@@ -37,8 +39,10 @@
...
-- (nonnull instancetype)initWithFileName:(nullable NSString *)fileName
-                          attachmentData:(nonnull NSData *)data
-                             contentType:(nonnull NSString *)contentType;
+- (instancetype)initWithFileName:(nullable NSString *)fileName
+                          attachmentData:(NSData *)data
+                             contentType:(NSString *)contentType;
 
 @end
+
+NS_ASSUME_NONNULL_END
```
Comment (TroubleMakerBen) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorBinaryAttachment.h]: Same as in `MSErrorAttachment`. =)

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorAttachment.h (hunk 5 above)
Comment (erychagov) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorAttachment.h]: When I open Xcode I see, that MSErrorAttachment.h is in folder "MobileCenterCrashes", but physically it is in folder "Model". And MSErrorAtachment.h is a public header.

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorBinaryAttachment.h (hunk 6 above)
Comment (erychagov) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorBinaryAttachment.h]: Same as in `MSErrorAttachment.h` :)

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorAttachment.h (hunk 5 above)
Comment (erychagov) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorAttachment.h]: I can move the file from `Model` to `MobileCenterCrashes` (physically). Or I can move the file from group `MobileCenterCrashes` to group `Model`.

File: MobileCenter/MobileCenter/Internals/Channel/MSChannel.h [hunk 7]
```diff
@@ -34,8 +34,8 @@ NS_ASSUME_NONNULL_BEGIN
  *
  * @return A new `MSChannelDefault` instance.
  */
-- (instancetype)initWithSender:(id<MSSender>)sender
-                       storage:(id<MSStorage>)storage
+- (instancetype)initWithSender:(nullable id<MSSender>)sender
```
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Channel/MSChannel.h]: For those two I think I would rather have nonnull sender and storage here and in the logmanager properties. I don't think of one use case where those can be null.

File: MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m [hunk 8]
```diff
@@ -97,15 +97,22 @@ + (nullable NSData *)dataForFile:(MSFile *)file {
   NSFileManager *fileManager = [NSFileManager defaultManager];
 
   // Check validity.
-  if (!directoryPath || !fileExtension || ![fileManager fileExistsAtPath:directoryPath]) {
+  if (!directoryPath || !fileExtension) {
+    return nil;
+  }
+
```
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m]: You can remove this new line. Basically our convention doesn't allow new lines within code blocks if it's not preceded by a comment.

File: MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorAttachment.h (hunk 5 above)
Comment (clpolet) [inline MobileCenterCrashes/MobileCenterCrashes/Internals/Model/MSErrorAttachment.h]: Maybe create another "physical" folder `Model` under `MobileCenterCrashes` for public model classes. If you do so take a look at other places were this rule may apply for consistency.

**Comment by TroubleMakerBen** [comment]
LGTM

**Comment by jaelim-ms** [comment]
I fixed the build. There were some header files exposed to public from Crashes framework which we hid previously.
